docs/versioning.md

```diff
@@ -0,0 +1,42 @@
+# Kubernetes API and Release Versioning
+
+Legend:
+
+* **Kube &lt;major&gt;.&lt;minor&gt;.&lt;patch&gt;** refers to the version of Kubernetes that is released. This versions all components: apiserver, kubelet, kubectl, etc.
+* **API vX[betaX]** refers to the version of the HTTP API.
+
+## Release Timeline
+
+### Minor version timeline
+
+* Kube 1.0
+* Kube 1.0.x: We create a 1.0-patch branch and backport critical bugs and security issues to it. Patch releases occur as needed.
+* Kube 1.1-alpha1: Cut from HEAD, smoke tested and released two weeks after Kube 1.0's release. Every two weeks a new alpha is released from HEAD.
+* Kube 1.1-beta1: When HEAD is feature complete, we create a 1.1-snapshot branch and release it as a beta. (The 1.1-snapshot branch may be created earlier if something that definitely won't be in 1.1 needs to be merged to HEAD.) This should occur 6-8 weeks after Kube 1.0. Development continues at HEAD and only fixes are backported to 1.1-snapshot.
+* Kube 1.1-rc1: Released from 1.1-snapshot when it is considered stable and ready for testing. Most users should be able to upgrade to this version in production.
+* Kube 1.1: Final release. Should occur between 3 and 4 months after 1.0.
+
+### Major version timeline
+
+There is no mandated timeline for major versions. They only occur when we need to start the clock on deprecating features. A given major version should be the latest major version for at least one year from its original release date.
+
+## Upgrade policies
+
+* Users can upgrade from any Kube 1.x release to any other Kube 1.x release as a rolling upgrade across their cluster.
```

Comment:
#4855 says:

> apiserver of version n.x must be accessible to a kubelet of version n.y if x>=y (TBD: actually, this section should just defer to the results of #4833)

and

>  kubelet version tuple must be reported for the cross-product of, at least, (kubelet, docker, kernel).

I don't think either of those conflict with this, and that the first statement does in fact imply that a rolling upgrade is possible? In any case I updated the language, please take a look.

Additionally, I do not think it's okay to require a stop-the-world upgrade to go from 1.x to 1.x+y. That's simply unacceptable for a production system.